### DISCUSSION TITLE
Why does the Sprite disappear after writing the Bluetooth code in the esp32?

### DISCUSSION THREAD

**Discussion by ENE4546** [Bodmer/TFT_eSPI#2972] [Q&A - General]
`BluetoothSerial SerialBT; 
void Bluetooth_Event(esp_spp_cb_event_t event, esp_spp_cb_param_t *param);
int state;`
Write this line of code, and the eye pattern will disappear,turn white screen.

**Comment by Bodmer** [comment]
I am not sure how you expect me to help you! You need to describe the problem in much more detail and provide example code. It is likely the problem is elsewhere in the sketch!  For instance, this works for me:
```
#include <TFT_eSPI.h>
TFT_eSPI tft = TFT_eSPI();
TFT_eSprite spr = TFT_eSprite(&tft);

#include <BluetoothSerial.h>
BluetoothSerial SerialBT;
void Bluetooth_Event(esp_spp_cb_event_t event, esp_spp_cb_param_t *param);
int state;

void setup() {
  tft.init();
  tft.fillScreen(TFT_BLACK);
  spr.setColorDepth(8);
  spr.createSprite(100, 50);
  spr.fillSprite(TFT_BLUE);
  spr.drawLine(0, 0, 99, 49, TFT_WHITE);
  spr.drawLine(0, 49, 99, 0, TFT_WHITE);
  spr.pushSprite(0, 0);
}

void loop() {

}

```

**Comment by AlexMarambio** [comment]
Hey there, im currently working on a project that uses tft sprites, sprites that changes every loop. Also using bluetooth, but happens that using sprites the screen freezes, but the code is still runing on the back. I change the sprite by only the tft objects and it works apparently correct. Maybe is something like that happening here, idk why but when making an animation with sprites the bluetooth serial and BLE freeze the display (Im using ttgo tdisplay 16mb)

**Reply by AlexMarambio** [reply]
I come with some updates, apparently it was the way i was working the sprites. I used to have a sprite for every "image" on my screen and use the pushToSprite function. When using just tft class was a solution but there was a flicker on the screen.

THE SOLUTION TO MY ISSUE WAS:
Just use only one sprite for all your screen assets! avoid usig pushToSprite, just make your code print or pushImages in the correct order :).
